File: code2/strRev.cpp
/**
 * REVERSING A STRING
 * author: Rajdeep Roy Chowdhury<rrajdeeproychowdhury@gmail.com> 
 * github: https://github.com/razdeep
 */
#include<iostream>

using namespace std;

string strRev(const string& s)
{
    static string temp=s;
    int k=0;
    for(int i=s.size()-1;i>=0;i--)
    {
        temp[k++]=s[i];
    }
    return temp;
}

int main()
{
    string s="Hello";
    string rev=strRev(s);
    cout<<rev<<endl;
    return 0;
}

File: code2/strRev.java
/*
 * strRev
 * author: Ashwek
 * github: @ashwek
 */

import java.lang.StringBuffer;

class strRev{

	public static void main(String args[]){
		String a = "Hello World";
		System.out.println("Orignal String = " + a);

		String rev = new StringBuffer(a).reverse().toString();
		System.out.println("Reverse String = " + rev);

	}
}
